components/ai_chat/resources/page/api/index.ts

```diff
@@ -112,32 +116,12 @@ class API {
     })
   }
 
-  addStateChangeListener(callback: (event: UIStateChangeEvent) => void) {
-    this.eventTarget.addEventListener('uistatechange', callback)
-  }
-
-  removeStateChangeListener(callback: (event: UIStateChangeEvent) => void) {
-    this.eventTarget.removeEventListener('uistatechange', callback)
-  }
-
-  private dispatchDebouncedStateChange = debounce(() => {
-    console.debug('dispatching uistatechange event', {...this.state})
-    this.eventTarget.dispatchEvent(
-      new Event('uistatechange')
-    )
-  }, 0)
-
-  public setPartialState(partialState: Partial<State>) {
-    this.state = { ...this.state, ...partialState }
-    this.dispatchDebouncedStateChange()
-  }
-
   private async getCurrentPremiumStatus() {
     const { status } = await this.service.getPremiumStatus()
```

Comment:
no because it's only for the initial "unknown" state. I don't know why it doesn't use `undefined` for that, but I don't really want to change it here.